src/validation.cairo

```diff
@@ -302,4 +307,41 @@ mod tests {
         let result = validate_proof_of_work(@10_u256, @block);
         assert!(result.is_ok(), "Expect prev block hash lt target");
     }
+
+
+    #[test]
+    fn test_compute_block_reward() {
+        let max_halvings: u32 = 64;
+        let reward_initial: u32 = REWARD_INITIAL.try_into().unwrap();
+        let halving_block_range = 210_000; // every 210 000 blocks
+        let mut nprevious_subsidy = REWARD_INITIAL * 2;
+        let mut loop_index: u32 = 0;
+        loop {
```

Comment:
Have mercy, keep it as simple as possible. Just your function a few times in a row, compare it to the expected value.

Reply:
I've fix a merge conflict issue and clean a bit.

I redo the same test for all halvings possible like this one: 

https://github.com/bitcoin/bitcoin/blob/0f68a05c084bef3e53e3f549c403bc90b1db319c/src/test/validation_tests.cpp#L24

Reply:
Please add simple test first.

Reply:
I have added few simple test too